### DISCUSSION TITLE
Do BitmapLayers lose their transparency when using DeckLayer and ArcGIS?

### DISCUSSION THREAD

**Discussion by corporealshift** [visgl/deck.gl#7885] [Q&A]
hey all, i'm trying to use [deck.gl](http://deck.gl/) with arcGIS (which is working fine) but i'm running into an issue with tile layers where the rendered bitmaps are not actually being transparent? When I add two tile layers (one just renders a circle) I can not see through the circle layer to the one below it. If I inverse the order I see the other layer and do not see the circle layer. I know that the circle layer has transparency (if I use the same layer with google maps it works as I would expect). Can anyone give me some direction on where to look, or what options I may need to change? I've found a few github threads but nothing that has yielded much results. Here is my tile layer definition which is used for both layers:
```
new TileLayer({
  id: layer.id,
  data: tileURL(layer, apiToken),
  maxZoom,
  minZoom,
  tileSize: 256,
  opacity: 1,
  // deck.gl expects us to render the returned bitmap (tile) ourselves, so create a new
  // BitmapLayer and return it with the returned image data.
  renderSubLayers: (props) => {
    const {
      bbox: {
        west,
        south,
        east,
        north,
      },
    } = props.tile;
    return new BitmapLayer(props, {
      data: null,
      image: props.data,
      bounds: [west, south, east, north],
    });
  },
});
```
and I'm using the DeckLayer object from the @deck.gl/arcgis library:
```
const deckLayer = new DeckLayer({
    'deck.layers': pluginLayers,
    'deck.getCursor': ({ isDragging }) => (isDragging ? 'grabbing' : 'default'),
  });
  const map = new ArcGISMap({
    basemap: 'arcgis-topographic',
    layers: [deckLayer],
  });
```
Is there something I'm missing? If I just use the tile layers in deck.gl directly (without arcgis) then everything works as I would expect.

**Comment by ibgreen** [comment]
I believe ArcGIS creates a WebGL context with precomposed alpha, it is just a hunch that this is related - perhaps our shader doesn't detect a zero alpha channel so it can't discard the transparent pixel and the z buffer gets touched. Unfortunately, the deck.gl team doesn't have much experience with precomposed alpha.

**Reply by corporealshift** [reply]
so am I better off just setting my TileLayers/bitmap layers directly through arcgis and skipping deckgl? The tool I'm working on has a variety of layer options available to users so we can continue to use deck.gl to render other layer types (like geojson)

**Reply by ibgreen** [reply]
Possibly - If this bug blocks you from achieving what you want. I don't have enough experience with the ArcGIS basemap rendering to say for sure. 

This module used to be maintained by Esri engineers. @damix911 are there any Esri eyes on the deck.gl ArcGIS module these days?

**Comment by icf-ekillops** [comment]
I'm encountering a similar issue where all my layers within the DeckLayer seem to be rendered at the exact same z-index and transparent features do not show any features below them
